src/material-experimental/mdc-dialog/dialog-container.ts

```diff
@@ -0,0 +1,176 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {FocusTrapFactory} from '@angular/cdk/a11y';
+import {DOCUMENT} from '@angular/common';
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  Inject,
+  NgZone,
+  OnDestroy,
+  Optional,
+  ViewEncapsulation
+} from '@angular/core';
+import {MatDialogConfig, MatDialogContainerBase} from '@angular/material/dialog';
```

Comment:
In other places we prefix the name of these base classes with an underscore so people don't end up consuming them.